src/main/kotlin/org/rust/ide/annotator/fixes/AddStructFieldsFix.kt

```diff
@@ -48,15 +50,7 @@ class AddStructFieldsFix(
         val body = structLiteral.structLiteralBody
         val fieldsToAdd = calculateMissingFields(body, decl)
         val defaultValueBuilder = RsDefaultValueBuilder(decl.knownItems, body.containingMod, RsPsiFactory(project), recursive)
-        val firstAdded = defaultValueBuilder.fillStruct(
-            body,
-            decl.fields,
-            fieldsToAdd
-        )
-
-        if (editor != null && firstAdded != null) {
-            val expr = firstAdded.expr
-            if (expr != null) editor.caretModel.moveToOffset(expr.textOffset)
-        }
+        val addedFields = defaultValueBuilder.fillStruct(body, decl.fields, fieldsToAdd)
+        editor?.buildAndRunTemplate(body, addedFields.mapNotNull { it.expr?.createSmartPointer() })
```

Comment:
If you don't want to invoke `SmartPointerManager.getInstance(project)` (from `createSmartPointer` extension method) on every `PsiElement` here, you can get it once like this:
```Kotlin
val initializers = with(SmartPointerManager.getInstance(project)) {
    addedFields.mapNotNull { it.expr }.map(::createSmartPsiElementPointer)
}
editor?.buildAndRunTemplate(body, initializers)
```
The same about `DestructureIntention`